geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/BinAggregatingIterator.scala

```diff
@@ -253,17 +175,61 @@ class BinAggregatingIterator extends SortedKeyValueIterator[Key, Value] with Log
    * Writes a bin record from a feature that has a arbitrary geometry.
    * A single internal point will be written.
    */
-  def writeGeometry(sf: KryoBufferSimpleFeature): Unit =
-    writeBinToBuffer(sf, sf.getAttribute(geomIndex).asInstanceOf[Geometry].getCentroid)
+  def writeGeometry(sf: KryoBufferSimpleFeature, byteBuffer: ByteBuffer): Unit =
+    writeBinToBuffer(sf, sf.getAttribute(geomIndex).asInstanceOf[Geometry].getCentroid, byteBuffer)
 
   /**
    * Writes geom + label
    */
-  def writeGeometryWithLabel(sf: KryoBufferSimpleFeature): Unit = {
-    writeGeometry(sf)
-    writeLabelToBuffer(sf)
+  def writeGeometryWithLabel(sf: KryoBufferSimpleFeature, byteBuffer: ByteBuffer): Unit = {
+    writeGeometry(sf, byteBuffer)
+    writeLabelToBuffer(sf, byteBuffer)
   }
 
+  override def deepCopy(env: IteratorEnvironment): SortedKeyValueIterator[Key, Value] = ???
```

Comment:
Yeah, if it ever gets called it would throw a NotImplementedError - but it would only get called if some other iterator is stacked on top of it, which doesn't happen right now... we can always go back and implement it later if we need to.